src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -289,12 +294,35 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                 page = JsonParser.toJava(response, BitbucketServerPullRequests.class);
                 pullRequests.addAll(page.getValues());
             }
+
+            AbstractBitbucketEndpoint endpointConfig = BitbucketEndpointConfiguration.get().findEndpoint(baseURL);
+            if (endpointConfig instanceof BitbucketServerEndpoint && ((BitbucketServerEndpoint)endpointConfig).isCallCanMerge()) {
```

Comment:
Can you add a code comment here explaining why this code is setting a field (with an explicit HTTP call) which is never used anywhere else in the code? (I have the context becuase I read the PR comments, but future devs could only have the code itself).